### PR TITLE
Put secret and oauth credentials into the private headers for now.

### PR DESCRIPTION
**Pull request by Yue-Wang-Google**
Removing this from the public API for the Firebase 6 release.

### CODE REVIEW

**Comment by Yue-Wang-Google** [comment]
Done.

**Comment by Yue-Wang-Google** [comment]
Done

File: Firebase/Auth/Source/Public/FIRAuthDataResult.h [hunk 1]
```diff
@@ -38,14 +37,7 @@ NS_SWIFT_NAME(AuthDataResult)
...
  */
-@property(nonatomic, readonly, nullable) FIRUser *user;
-
-/** @property credential
-    @brief The updated OAuth credential after the the sign-in, link and reauthenticate action.
-    @detial This property is for OAuth sign in only.
- */
-@property(nonatomic, readonly, nullable) FIROAuthCredential *credential;
-
+@property(nonatomic, readonly) FIRUser *user;
```
Comment (paulb777) [inline Firebase/Auth/Source/Public/FIRAuthDataResult.h]: Isn't making `user` nullable a bugfix that should stay?

File: Firebase/Auth/Source/Auth/FIRAuthDataResult_Internal.h [hunk 2]
```diff
@@ -16,10 +16,18 @@
...
 
+@class FIROAuthCredential;
+
 NS_ASSUME_NONNULL_BEGIN
 
 @interface FIRAuthDataResult () <NSSecureCoding>
 
+/** @property credential
+    @brief The updated OAuth credential after the the sign-in, link and reauthenticate action.
+    @detial This property is for OAuth sign in only.
```
Comment (paulb777) [inline Firebase/Auth/Source/Auth/FIRAuthDataResult_Internal.h]: detail - spelling

**Comment by Yue-Wang-Google** [comment]
As I think more about it, I believe following the existing API protocol  (user is not null) is better. I read all the code that generates the FIRAuthDataResult, and  believes changing the implementation to ensure user is not null  makes more sense. I explain my reasoning in  the commit : https://github.com/firebase/firebase-ios-sdk/pull/2916/commits/e78260ffd415b60a4c04c1f0175797b8128db86e  So please take a look and see if this looks good to you guys. Thanks

File: Firebase/Auth/Source/Public/FIRAuthDataResult.h (hunk 1 above)
Comment (Yue-Wang-Google) [reply Firebase/Auth/Source/Public/FIRAuthDataResult.h]: See below and the message of the 3rd commit for my explanation and please let me know what do you believe would be the better way to resolve this.

**Comment by renkelvin** [comment]
@Yue-Wang-Google Could you create a new pr to expose these fields?

**Comment by Yue-Wang-Google** [comment]
see https://github.com/firebase/firebase-ios-sdk/pull/3089

**Pull request merged**